### DISCUSSION TITLE
Allow overriding keyPrefix in the TFunction

### DISCUSSION THREAD

**Discussion by ericfennis** [i18next/i18next#1802] [Ideas]
I opened issue in #1801, but @adrai was not convinced yet so he closed it. So I opened this discussion thread to see if maybe more devs wants this. So leave an upvote if you agree.

## 🚀 Feature Idea

Allow overriding the keyPrefix in the TFunction that is returned by the `getFixedT` function from the instance. This one used in the react-i18next useTranslation function.

## Motivation

It makes it possible to use a specific translation in a different namespace, but without the keyPrefix.
Currently it is possible to override the namespace in the T-function but would be nice to override the `keyPrefix` as well.
The problem is when you using a keyPrefix in the useTranslation hook 

## Example

```js
t('cancel', 'Cancel', { ns: 'common', keyPrefix: 'buttons' })
```

In a React component

```jsx
const Component = () => {
  const { t } = useTranslation('common', { keyPrefix: 'content' })

  return (
    <div>
      { /* Setting the keyPrefix option to undefined or something else, will override the keyPrefix defined in the useTranslation hook, so no keyPrefix will be added to select the translation. */}
      { /* So no keyPrefix will be added to select the translation. */}
      <h1>
      { t('helloWorld', 'Hello World', { ns: 'common', keyPrefix: 'buttons' }) // path: `common:buttons.hello`
      }
      </h1>

      <p>
      { t('body', 'Welcome to this website!') // path: `common:content.body`
      }
      </p>

    </div>
  )
}
```

## Alternative solution what works now (but is not nice)

```jsx
const Component = () => {
  const { t } = useTranslation('common', { keyPrefix: 'content' })
  const { t: commonButtonsT } = useTranslation('common')

  return (
    <div>
      <h1>
      { commonButtonsT('helloWorld', 'Hello World') }
      </h1>

      <p>
      { t('body', 'Welcome to this website!') }
      </p>

    </div>
  )
}
```

**Comment by adrai** [comment]
Sorry, but what I still don't get is:
Why should `t('cancel', 'Cancel', { ns: 'common', keyPrefix: 'buttons' })` be better than `t('buttons.cancel', 'Cancel', { ns: 'common' })`?
